src/main/resources/org/jenkinsci/plugins/docker/swarm/DockerSwarmAgentTemplate/config.jelly

```diff
@@ -10,10 +10,10 @@
         <f:textbox value="${dockerSwarmAgentTemplate.image}" default="java:8"/>
     </f:entry>
     <f:entry title="Unix Command" field="unixCommand">
-        <f:textarea value="${dockerSwarmAgentTemplate.unixCommand}" default="sh&#10;-cx&#10;curl --connect-timeout 20 --max-time 60 -o agent.jar $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JAR_URL &amp;&amp; java -jar agent.jar -jnlpUrl $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JNLP_URL -secret $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_SECRET -noReconnect -workDir /tmp"/>
+        <f:textarea value="${dockerSwarmAgentTemplate.unixCommand}" default="sh&#10;-cx&#10;curl --connect-timeout 20 --max-time 60 -o agent.jar $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JAR_URL &amp;&amp; java -classpath agent.jar hudson.remoting.jnlp.Main -headless -url $DOCKER_SWARM_PLUGIN_JENKINS_URL -noreconnect -workDir /tmp $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_SECRET $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_NAME"/>
```

Comment:
Pending this PR, you can replace `$DOCKER_SWARM_PLUGIN_JENKINS_URL` with `${DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JAR_URL%%jnlpJars/agent.jar}`

Reply:
Can or should?

Reply:
Sorry, to rephrase & expand: if you are running 1.11 but wish to access Jenkins HTTP via a different host than the canonical root URL, for example because you are running in Kubernetes and Swarm is inside the network and connections to the Jenkins controller should not exit the network and then come back in via ingress, then it is possible to configure that using a command like the one above _but_ since `$DOCKER_SWARM_PLUGIN_JENKINS_URL` is not defined you would need to obtain the URL another way, such as with shell scripting `${DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JAR_URL%%jnlpJars/agent.jar}`. When 1.12 defined the new variable, then you could switch to the simpler form suggested here.